### PR TITLE
Implement Set methods from ECMAScript Specification Features/#4128

### PR DESCRIPTION
**Pull request by Hemenguelbindi**
This Pull Request fixes/closes #{issue_num}.

It changes the following:
- Implemented the Set methods from the ECMAScript specification
    - Set.prototype.difference(other)
    - Set.prototype.intersection(other)
    - Set.prototype.isDisjointFrom(other)
    - Set.prototype.isSubsetOf(other)
    - Set.prototype.isSupersetOf(other)
    - Set.prototype.symmetricDifference(other)
    - Set.prototype.union(other)
- Added test cases for all newly implemented methods to ensure specification compliance.
- Updated relevant documentation and comments for clarity.

Additional Information:
This implementation aligns with the ECMAScript specification, section 24.2.4, and follows the behavior described in the [Set Methods Proposal](https://github.com/tc39/proposal-set-methods). Test coverage includes edge cases such as empty sets, overlapping sets, and disjoint sets.

Feel free to let me know if there are areas for improvement or adjustments!

### CODE REVIEW

**Comment by nekevss** [comment]
This is looking great so far!

It looks like you may need to run `cargo fmt --all`

Also, if you remove `set-methods` from the `test262_config.toml`, then you should be able to run the `Set` built-in test262 suite with `boa_tester`:

```bash
cargo run --release --bin boa_tester -- run -vv -s test/built-ins/Set/prototype
```

If you want more verbose feedback, you can run:

```bash
cargo run --release --bin boa_tester -- run -vvv -s test/built-ins/Set/prototype
```

Adding the specific method on the end of the path can target that specific method.

**Comment by Hemenguelbindi** [comment]
i am done all fix need this issus https://github.com/boa-dev/boa/issues/4128

**Review by nekevss — CHANGES_REQUESTED** [review]
This is looking great so far!

Have some general feedback to help increase test coverage on the implementation and a typo.

File: core/engine/src/builtins/set/mod.rs [hunk 1]
```diff
@@ -491,6 +500,418 @@ impl Set {
...
+            .and_then(JsObject::downcast_ref::<OrderedSet>)
+        else {
+            return Err(JsNativeError::typ()
+                .with_message("Method Set.prototype.difference called on incompatible receiver")
+                .into());
+        };
+        // 3. Let otherRec be ? GetSetRecord(other).
+        //    (ECMAScript 2022, 24.2.3.6 step 3)
+        let other = args.get_or_undefined(0);
+        let Some(other_set) = other
```
[the remark this answers is not in the document]
Comment (Hemenguelbindi) [reply core/engine/src/builtins/set/mod.rs:532]: I understand you. I need to implement the equivalent of GetRecord. I think I should do this in the orderset file, and it will be called SetRecord, which will eliminate all the repetitive code I wrote before, right?

[the remark this answers is not in the document]
Comment (nekevss) [reply core/engine/src/builtins/set/mod.rs:532]: The method that needs to be implemented is [here](https://tc39.es/ecma262/#sec-getsetrecord). My initial reaction based off reading is that this abstract op would be implemented in `mod.rs`.

**Comment by Hemenguelbindi** [comment]
I'm sorry that I've been missing for so long, there were too many blockages at work, so I fixed everything according to your recommendations, and also divided the test logic for some tests.

**Review by nekevss — APPROVED** [review]
Thanks for working on this!!!
